api/v1alpha1/initialbuildpipelineselector_types.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	tektonapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+// Defines boolean value in a string to have not set as well.
+// +kubebuilder:validation:Enum=true,false,True,False,1,0
+type BoolString string
+
+const (
+	Bool_true  BoolString = "true"
+	Bool_false BoolString = "false"
+	Bool_True  BoolString = "True"
+	Bool_False BoolString = "False"
+	Bool_1     BoolString = "1"
+	Bool_0     BoolString = "0"
+)
+
+func IsBoolStringTrue(arg BoolString) bool {
+	return arg == Bool_true || arg == Bool_True || arg == Bool_1
+}
+
+// WhenConditions defines requirements when specified build pipeline must be used.
+// All conditions are connected via AND, whereas cases within any condition connected via OR.
+// Example:
+//
+//	language: java
+//	projectType: spring,quarkus
+//	annotations:
+//	   builder: gradle,maven
```

Comment:
So this means you can't match labels with a comma in the value? Should this be `map[string][]string` instead?

Reply:
AFAIK, [it's not allowed](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) to have comma in the label/annotation key/value. So, shouldn't be a problem.
I was thinking about string array (explained in [one of previous comments](https://github.com/redhat-appstudio/build-service/pull/84#discussion_r1039714775)).